src/taco-utils/cordovaHelper.ts

```diff
@@ -158,18 +159,27 @@ module TacoUtility {
             return cordovaArgs.concat(additionalArguments);
         }
 
-        public static toCordovaRunArguments(commandData: Commands.ICommandData, platforms: string[] = null): Cordova.ICordovaRawOptions {
-            // Run, build, emulate, prepare and compile all use the same format at the moment
+        public static toCordovaRunArguments(cordovaVersion: string, commandData: Commands.ICommandData, platforms: string[] = null): Cordova.ICordovaRawOptions | Cordova.ICordova540RawOptions {
+            if (semver.gte(cordovaVersion, CordovaHelper.RAW_API_540_VERSION)) {
+                return CordovaHelper.toCordovaRaw540Arguments(commandData, platforms);
+            }
+
             return CordovaHelper.toCordovaArgumentsInternal(commandData, platforms);
         }
 
-        public static toCordovaBuildArguments(commandData: Commands.ICommandData, platforms: string[] = null): Cordova.ICordovaRawOptions {
-            // Run, build, emulate, prepare and compile all use the same format at the moment
+        public static toCordovaBuildArguments(cordovaVersion: string, commandData: Commands.ICommandData, platforms: string[] = null): Cordova.ICordovaRawOptions | Cordova.ICordova540RawOptions {
+            if (semver.gte(cordovaVersion, CordovaHelper.RAW_API_540_VERSION)) {
+                return CordovaHelper.toCordovaRaw540Arguments(commandData, platforms);
+            }
+
             return CordovaHelper.toCordovaArgumentsInternal(commandData, platforms);
         }
 
-        public static toCordovaTargetsArguments(commandData: Commands.ICommandData, platforms: string[] = null): Cordova.ICordovaRawOptions {
-            // Run, build, emulate, prepare and compile all use the same format at the moment
+        public static toCordovaTargetsArguments(cordovaVersion: string, commandData: Commands.ICommandData, platforms: string[] = null): Cordova.ICordovaRawOptions | Cordova.ICordova540RawOptions {
```

Comment:
Is there a reason you removed the comment? And did you consider putting the choice between 5.4.0+ and < 5.4.0 argument style in toCordovaArgumentsInternal?